### ISSUE TITLE
`Style/SafeNavigation` False positive with `TargetRubyVersion: 2.0`

### ISSUE THREAD

**Issue by mishina2228** [rubocop/rubocop#10679]
## Expected behavior

No offense with `Style/SafeNavigation`,
because safe navigation operator was introduced in Ruby 2.3.

## Actual behavior

```
$ rubocop --debug sample.rb
For /tmp/rubocop: configuration from /tmp/rubocop/.rubocop.yml
Default configuration from /home/owner/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rubocop-1.30.0/config/default.yml
Use parallel by default.
Running parallel inspection
Loading cache from /home/owner/.cache/rubocop_cache/d10146b520466ece109729e7403f11d5d0a5ad94/6d7a3b621ca1730e04accd938619e4bdab66cfb1/59ebe8899ab1693cd0e5ef08914c6351047e072e
Inspecting 1 file
Scanning /tmp/rubocop/sample.rb
Loading cache from /home/owner/.cache/rubocop_cache/d10146b520466ece109729e7403f11d5d0a5ad94/6d7a3b621ca1730e04accd938619e4bdab66cfb1/59ebe8899ab1693cd0e5ef08914c6351047e072e
C

Offenses:

sample.rb:2:3: C: [Correctable] Style/SafeNavigation: Use safe navigation (&.) instead of checking if an object exists before calling the method.
  arg && arg.bar
  ^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.11757984200085048 seconds
```

## Steps to reproduce the problem

`.rubocop.yml`
```yml
AllCops:
  TargetRubyVersion: 2.0
  NewCops: enable
```

`sample.rb`
```ruby
def foo(arg)
  arg && arg.bar
end
```

## RuboCop version

```
$ rubocop -V
1.30.0 (using Parser 3.1.2.0, rubocop-ast 1.18.0, running on ruby 3.1.0 x86_64-linux)
```
